packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx

```diff
@@ -46,40 +45,43 @@ const LinkContent = React.forwardRef<HTMLAnchorElement, LinkContentProps>(
     const [linkUrl, setLinkUrl] = React.useState(link.url);
     const linkInputRef = React.useRef<HTMLInputElement>(null);
     const [showRemoveButton, setShowRemoveButton] = React.useState(false);
-    const [invalidLinkError, setInvalidLinkError] = React.useState('');
+    const [isSaveDisabled, setIsSaveDisabled] = React.useState(false);
 
     const handleOpenEditPopover: React.MouseEventHandler<HTMLAnchorElement> = (e) => {
       e.preventDefault();
       setPopoverOpen(true);
       setShowRemoveButton(true);
     };
 
-    const handleSave: React.FormEventHandler = (e) => {
-      e.stopPropagation();
+    const onLinkChange: React.FormEventHandler = (e) => {
+      setIsSaveDisabled(false);
+      setLinkUrl(e.target.value);
+
+      const delayValidation = setTimeout(() => {
```

Comment:
Why do we need the timeout? Aren't we able to run that logic directly on every update?

Reply:
ohh you're right!

Reply:
I think this type of validation is apt for search and so I tend to think and apply same logic 😒